src/ansys/simai/core/data/optimizations.py

```diff
@@ -332,27 +366,33 @@ def run_non_parametric(
                 )
             )
             optimization.wait()
+            trial_result = {"objectives": [], "geometries": []}
             logger.debug("Optimization defined. Starting optimization loop.")
-            iterations_results: List[Dict] = []
             with keep.running(on_fail="warn"):
                 for _ in range(n_iters):
                     logger.debug("Running iteration")
                     progress_bar.set_description("Running iteration")
                     trial_run = optimization._run_iteration({})
                     trial_run.wait()
-                    iteration_result = {
-                        "objective": trial_run.fields["outcome_values"],
-                    }
-                    progress_bar.set_postfix(**iteration_result)
+                    progress_bar.set_postfix(
+                        {
+                            "objective": trial_run.fields["outcome_values"],
+                        }
+                    )
                     if trial_run.fields.get("is_feasible", True):
-                        iterations_results.append(iteration_result)
+                        trial_result["objectives"].append(trial_run.fields["outcome_values"])
+                        trial_result["geometries"].append(trial_run.fields["geometry"])
                     else:
                         logger.debug("Trial run results did not match constraints. Skipping.")
                     logger.debug("Trial complete.")
                     progress_bar.update(1)
                 logger.debug("Optimization complete.")
                 progress_bar.set_description("Optimization complete.")
-            return iterations_results
+
+            optimization_result = OptimizationResult(
```

Comment:
raw dicts are horrible to work with, I can't review this, trusting you on this one